src/japi.c

```diff
@@ -337,6 +339,19 @@ japi_context *japi_init(void *userptr)
 	return ctx;
 }
 
+int japi_set_max_linebuf_size(japi_context *ctx, size_t max_linebuf_size_user)
+{
+	/* Error handling */
+	if (ctx == NULL) {
+		fprintf(stderr, "ERROR: JAPI context is NULL.\n");
+		return -1;
+	}
+
```

Comment:
Maybe we add some error handling here. Too small values should not be allowed. Maybe use 64MiB as minimum value and warn if user wants to set something smaller? Not sure if an upper bound makes sense.

```suggestion
	if (max_linebuf_size_user < 1024*1024){
		fprintf(stderr, "ERROR: max_linebuf_size cannot be set to a value less than 1024*1024.")
		return -2;
	}

```

This suggested change requires adding -2 to the return values in .h!

Reply:
Done. Minimum max line size is CREADLINE_BLOCK_SIZE (1024 Bytes)